### PR TITLE
Bugfix/ingest to embed message exchange

### PR DESCRIPTION
**Pull request by gecBurton**
## Context

In https://github.com/i-dot-ai/redbox-copilot/pull/108 I erroneously introduced an error where I put a  `Chunk` rather than a `EmbedQueueItem` message into the embed queue

## Changes proposed in this pull request

1. in the `ingest` app i now put an `EmbedQueueItem` onto the embed queue
2. I have also simplified the emebed app

## Guidance to review

```commandLine
docker compose down
docker compose build ingest embed
docker compose up -d  elasticsearch minio rabbitmq ingest embed core-api
```

upload a file via: http://0.0.0.0:5002/docs#/file/create_upload_file_file_post

watch http://localhost:15672/#/queues and see that:
1. the ingest queue grows to 1 message
2. the ingest queue falls back to 0 messages and the embed queue grows
3. the embed queue shrinks as the embed worker processes `docker compose logs -f emebd` 

## Relevant links

n/a

## Things to check

- [ ] I have added any new ENV vars in all deployed environments
- [ ] I have tested any code added or changed

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -163,6 +163,7 @@ services:
       - elasticsearch
     restart: unless-stopped
   rabbitmq:
+    user: "423642704:849146750"
```
Comment (gecBurton) [inline docker-compose.yml:166]:
```suggestion
```

File: ingest/src/app.py [hunk 2]
```diff
@@ -59,11 +59,12 @@ def ingest_file(self, file: File):
         logging.info(f"written {len(items)} chunks to elasticsearch")
 
         for chunk in chunks:
+            queue_item = EmbedQueueItem(model=env.embedding_model, sentence=chunk.text)
```
Comment (gecBurton) [inline ingest/src/app.py:62]: this is the key change

File: embed/src/app.py [hunk 3]
```diff
@@ -28,28 +27,20 @@
...
-    # Check to see if we run in polling mode from a queue
-    if "QUEUE_URI" in os.environ:
-        log.info("Queue URI found, starting queue listener.")
-        queue_uri = os.environ["QUEUE_URI"]
-        queue_name = os.environ.get("EMBED_QUEUE_NAME", "redbox-embed-queue")
-        queue_poll_interval = int(os.environ.get("QUEUE_POLL_INTERVAL", 5))
-
-        poll_thread = threading.Thread(target=poll_queue_every, args=(queue_uri, queue_name, queue_poll_interval))
-        poll_thread.start()
+    thread = threading.Thread(target=subscribe_to_queue)
```
Comment (gecBurton) [inline embed/src/app.py:37]: i think we always want to run this thread?

File: embed/src/app.py [hunk 4]
```diff
@@ -171,62 +161,24 @@ def embed_sentences(model: str, sentences: list[str]):
...
-        poll_queue(queue_uri, queue_name)
-        time.sleep(interval)
-
-
-def poll_queue(queue_uri: str, queue_name: str, max_connection_attempts: int = 10):
-    logging.info(f"Polling queue {queue_name}")
-
-    connection = None
-
-    for i in range(max_connection_attempts):
```
Comment (gecBurton) [inline embed/src/app.py:187]: `pika` already handles retries

File: embed/src/app.py [hunk 5]
```diff
@@ -171,62 +161,24 @@ def embed_sentences(model: str, sentences: list[str]):
     return output
 
 
-def poll_queue_every(queue_uri: str, queue_name: str, interval: int = 5):
```
Comment (gecBurton) [inline embed/src/app.py:174]: ive tested this IRL, there is no need to poll as pika handles this via the callback mechanism

File: embed/src/app.py [hunk 6]
```diff
@@ -171,62 +161,24 @@ def embed_sentences(model: str, sentences: list[str]):
...
-    output = embed_sentences(embed_queue_item.model, [embed_queue_item.sentence])
-    output = EmbeddingResponse(**output)
-
-    # TODO: Send the output to Elasticsearch?
-
-    logging.info(f"Embedding ID {output.embedding_id} complete for {method.delivery_tag}")
-    ch.basic_ack(delivery_tag=method.delivery_tag)
-    ch.stop_consuming()
+    finally:
+        ch.basic_ack(delivery_tag=method.delivery_tag)
```
Comment (gecBurton) [inline embed/src/app.py:184]: if this is not `basic_nack` then the unprocessed items remain in the queue.... tbh i wasnt sure what we want to do here?

**Review by lmwilkigov — APPROVED** [review]

**Pull request merged**